receiver/prometheusreceiver/internal/staleness_store.go

```diff
@@ -66,24 +78,40 @@ func (ss *stalenessStore) refresh() {
 
 // isStale returns whether lbl was seen only in the previous scrape and not the current.
 func (ss *stalenessStore) isStale(lbl labels.Labels) bool {
+	ss.mu.Lock()
+	defer ss.mu.Unlock()
+
 	hash := lbl.Hash()
-	return ss.previousHashes[hash] && !ss.currentHashes[hash]
+	_, inPrev := ss.previousHashes[hash]
+	_, inCurrent := ss.currentHashes[hash]
+	return inPrev && !inCurrent
 }
 
 // markAsCurrentlySeen adds lbl to the manifest of labels seen in the current scrape.
 // This method should be called before refresh, but during a scrape whenever labels are encountered.
-func (ss *stalenessStore) markAsCurrentlySeen(lbl labels.Labels) {
-	ss.currentHashes[lbl.Hash()] = true
+func (ss *stalenessStore) markAsCurrentlySeen(lbl labels.Labels, at int64) {
```

Comment:
The `at` argument is unclear. Please document or use a more narrow/specialized type to make the intent clear.

Reply:
`at` is the `time at which the point was obtained` but perhaps let me use `timeMs`